extensions/ql-vscode/package.json

```diff
@@ -1422,6 +1422,7 @@
     "update-vscode": "node ./node_modules/vscode/bin/install",
     "format": "prettier --write **/*.{ts,tsx} && eslint . --ext .ts,.tsx --fix",
     "lint": "eslint . --ext .js,.ts,.tsx --max-warnings=0",
+    "lint:markdown": "markdownlint-cli2-config \"../../.markdownlint-cli2.cjs\" \"**/*.{md,mdx}\" \"!node_modules\"",
```

Comment:
Can you also add this to the [`Actions workflow`](https://github.com/github/vscode-codeql/blob/c882a9fc1414f3e2c285251a8c54b3beada15606/.github/workflows/main.yml#L98-L106)? That will ensure that it actually gets checked in CI.

Also, will this lint our top-level `README.md` file? This is run from the `extensions/ql-vscode` subdirectory, so I'm not sure how that works with the top-level Markdown files.

Reply:
I'm not going to add to CI until we've fixed all violations. And I'm deliberately not covering the generated files in this PR since that's more complicated.

My plan is (as separate PRs):

1. Get this basic PR in
2. Fix generated files
3. Enable in CI

> Also, will this lint our top-level `README.md` file?

Good point. I'll look into this now. Having `package.json` in a subfolder makes configuration a bit trickier here!

Reply:
767f9934e95e99a65334442190afa6331296dbc6 changes the linting so that it checks the whole codebase; subsequent commits fix the additional violations.